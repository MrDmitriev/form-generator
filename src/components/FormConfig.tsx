import { ChangeEvent } from 'react'

type Props = {
	onConfigChange: (e: ChangeEvent<HTMLTextAreaElement>) => void,
	onConfigSubmit: () => void,
	value: string
}
export default function FormConfig({onConfigChange, onConfigSubmit, value}: Props) {
	return (
		<div id="config" className="tabcontent active">
			<form className='form-configurator'>
				<textarea className='form-configurator__input' value={value} onChange={onConfigChange} placeholder="{items: []}" />
			</form>
			<button className='form-configurator__submit' onClick={onConfigSubmit}>Apply</button>
		</div>
	)
}
