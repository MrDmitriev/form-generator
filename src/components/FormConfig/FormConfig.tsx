import { ChangeEvent } from 'react'

type Props = {
	onConfigChange: (e: ChangeEvent<HTMLTextAreaElement>) => void,
	onConfigSubmit: () => void,
	value: string
}

// Enables to enter a valid json with form configuration
export default function FormConfig({onConfigChange, onConfigSubmit, value}: Props) {
	return (
		<div id="config" className="tabcontent active">
			<form className='form-configurator'>
				<textarea
					className='form-configurator__input'
					value={value}
					onChange={onConfigChange}
					placeholder="{'items': [], 'buttons': []}"
				/>
			</form>
			<button
				className='form-configurator__submit'
				onClick={onConfigSubmit}
			> Apply
			</button>
		</div>
	)
}
