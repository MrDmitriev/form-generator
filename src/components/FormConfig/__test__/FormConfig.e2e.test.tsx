import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';

import FormConfig from '../FormConfig';

Enzyme.configure({ adapter: new Adapter() });

describe('FormConfig component', () => {
	it(`should match snapshot`, () => {
		const onConfigChange = jest.fn();
		const onConfigSubmit = jest.fn();

		const props = {
			value: "",
			onConfigChange,
			onConfigSubmit
		};
	
		const wrapper = renderer.create(<FormConfig {...props} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});

	it('Should update form config after change a textarea', () => {
		const onConfigChange = jest.fn();
		const onConfigSubmit = jest.fn();

		const props = {
			value: "",
			onConfigChange,
			onConfigSubmit
		};

		const e = {
			currentTarget: {
				value: "{'items': [{'type': 'button'}]}"
			}
		};

		const wrapper = shallow(<FormConfig {...props} />);
		wrapper.find('textarea').first().simulate('change', e);
		expect(onConfigChange).toHaveBeenCalledTimes(1);
	})

	it('Should toggle to result tab after submitting config form', () => {
		const onConfigChange = jest.fn();
		const onConfigSubmit = jest.fn();

		const props = {
			value: "",
			onConfigChange,
			onConfigSubmit
		};

		const wrapper = shallow(<FormConfig {...props} />);
		wrapper.find('button').first().simulate('click');
		expect(onConfigSubmit).toHaveBeenCalledTimes(1);
	})
})