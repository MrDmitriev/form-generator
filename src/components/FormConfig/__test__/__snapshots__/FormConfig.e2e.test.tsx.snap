// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormConfig component should match snapshot 1`] = `
<div
  className="tabcontent active"
  id="config"
>
  <form
    className="form-configurator"
  >
    <textarea
      className="form-configurator__input"
      onChange={[MockFunction]}
      placeholder="{'items': [], 'buttons': []}"
      value=""
    />
  </form>
  <button
    className="form-configurator__submit"
    onClick={[MockFunction]}
  >
     Apply
  </button>
</div>
`;
